### PR TITLE
IGC parsing: parse long date / fallback to GPS altitude

### PR DESCRIPTION
**Pull request by juge2**
* date parsing works now also with long format of newer spec 'HFDTEDATE:'
* if barometric/pressure altitude is not available, fall back to GPS provided data

### CODE REVIEW

File: src/services/igc-service.js [hunk 1]
```diff
@@ -253,7 +261,13 @@ const igcService = {
...
    */
   getAltitudeFromBRecord(BRecord) {
-    return Number(BRecord.substr(25, 5));
+    let altitude = Number(BRecord.substr(25, 5));
+    if (altitude === 0) {
+      // fall back to GPS data
+      // ... well, for real pressure altitude of '0', it will be wrong but this should not happen too often ;-)
+      altitude = Number(BRecord.substr(30, 5));
+    }
+    return altitude;
```
Comment (kaplona) [inline src/services/igc-service.js:270]: I'm curious, do you have IGC tracks with pressure altitude reported as zeros?

In old IGC format gps altitude is reccommended and not mandatory, so some devices may not have it in `B` records. No need to worry, I will test on my old IGC files and make code changes if needed.

File: src/services/igc-service.js [hunk 2]
```diff
@@ -102,14 +102,22 @@ const igcService = {
...
-        dateRecord = record;
+      if (record.substring(2, 5) === 'DTE') {
+        if(record.substring(5, 9) === 'DATE') {
+          // new spec long format
+          dateRecord = record.substring(10,16);
+        }
+        else {
+          // original spec format
+          dateRecord = record.substring(5);
+        }
```
[the remark this answers is not in the document]
Comment (juge2) [reply src/services/igc-service.js:120]: OK - sure, got it, no problem ... will try to follow your lint rules

File: src/services/igc-service.js (hunk 1 above)
Comment (juge2) [reply src/services/igc-service.js:270]: Yes, it is zero with what I currently use: XCTrack on my mobile phone (without barometer sensor)  ... actually just for convenience to easily add my flight to my training log book (and koifly logs :-)).

here is a B entry:
B1806225218422N01305423EA000000005348

I can send you a full igc file, if you like.

File: src/services/igc-service.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaplona) [reply src/services/igc-service.js:120]: No, I didn't change it. You can either make another commit or click "Commit suggestion" button in my comment, it should just add a commit with suggested changes to your branch.

File: src/services/igc-service.js (hunk 1 above)
Comment (kaplona) [reply src/services/igc-service.js:270]: Interesting... Yes, could you please send me your file, I want to see if there's any indication that a device doesn't have a barometer.

File: src/services/igc-service.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaplona) [reply src/services/igc-service.js]: Actually I can commit it from github too.

**Pull request merged**